### PR TITLE
Special layer level counts

### PR DESCRIPTION
**Pull request by chrisforbes**
Improves #38.

### CODE REVIEW

**Comment by chrisforbes** [comment]
@tobine could you review this? It's pretty bad if we get stuck in these loops for 2^32 iterations, so I figure the next SDK probably wants this?

**Comment by tobine** [comment]
There are various "TODO : Do not iterate over every possibility" comments in the code related to looping over levelCount/layerCount. Can those now be removed or do you foresee further consolidation?

File: layers/draw_state.cpp [hunk 1]
```diff
@@ -7595,6 +7622,8 @@ VK_LAYER_EXPORT VKAPI_ATTR VkResult VKAPI_CALL vkGetSwapchainImagesKHR(
             image_node.layout = VK_IMAGE_LAYOUT_UNDEFINED;
             auto swapchain_node = dev_data->device_extensions.swapchainMap[swapchain];
             image_node.format = swapchain_node->createInfo.imageFormat;
+            image_node.mipLevels = 1;
+            image_node.arrayLayers = 1;
```
Comment (lentinem) [inline layers/draw_state.cpp]: Swapchain images can have multiple arrayLayers.

File: layers/draw_state.cpp [hunk 2]
```diff
@@ -4614,13 +4616,38 @@ VK_LAYER_EXPORT VKAPI_ATTR VkResult VKAPI_CALL vkCreateImage(VkDevice device, co
...
+
+
 VK_LAYER_EXPORT VKAPI_ATTR VkResult VKAPI_CALL vkCreateImageView(VkDevice device, const VkImageViewCreateInfo* pCreateInfo, const VkAllocationCallbacks* pAllocator, VkImageView* pView)
 {
     layer_data* dev_data = get_my_data_ptr(get_dispatch_key(device), layer_data_map);
     VkResult result = dev_data->device_dispatch_table->CreateImageView(device, pCreateInfo, pAllocator, pView);
     if (VK_SUCCESS == result) {
         loader_platform_thread_lock_mutex(&globalLock);
-        dev_data->imageViewMap[*pView] = unique_ptr<VkImageViewCreateInfo>(new VkImageViewCreateInfo(*pCreateInfo));
+        auto image_view = unique_ptr<VkImageViewCreateInfo>(new VkImageViewCreateInfo(*pCreateInfo));
```
Comment (lentinem) [inline layers/draw_state.cpp]: Is there a reason to store a pointer to a copy instead of just a copy?

Comment (chrisforbes) [inline layers/draw_state.cpp]: Not new, but we can revisit this..

**Comment by chrisforbes** [comment]
@tobine I think there's more work to do to make /all/ of those TODOs go away.

File: layers/draw_state.cpp (hunk 2 above)
Comment (lentinem) [inline layers/draw_state.cpp]: Yea I know; this seems like a good opportunity to change it though unless there is a reason not to. I'm not sure why pointers were used by some of these in the first place but if there is a good reason I'm ok with it.

**Comment by chrisforbes** [comment]
Updated to fix swapchain image layout count.

**Comment by tobine** [comment]
Mark, can you give this a quick test and push it? I've reviewed it an looks good, but I'm unable to test at the moment.

**Comment by mark-lunarg** [comment]
Looked fine on Linux and Windows -- pushed to Master.

**Pull request closed without merging**